staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go

```diff
@@ -309,6 +309,26 @@ func IsValidIP(value string) []string {
 	return nil
 }
 
+// IsValidIPv4Address tests that the argument is a valid IPv4 address.
+func IsValidIPv4Address(fldPath *field.Path, value string) field.ErrorList {
+	var allErrors field.ErrorList
+	ip := net.ParseIP(value)
```

Comment:
k8s/utils/net has a short cut function. I would strongly advise to replace this with
`
parse() => error result into invalid ip


netutils.IsIPv4()=> error result into invalid ipv4
`

Reply:
So I tried that out, but it may have had some unintended consequences. It added `k8s.io/utils` as a dependency of `k8s.io/apimachinery`, I'm not actually sure if that's an issue or not. The diff is in this commit: https://github.com/robscott/kubernetes/commit/9e282ee078cc817e350622401219028e2e77eaa1. @liggitt Do you know if adding `k8s.io/utils` as a dependency of `k8s.io/apimachinery` is something we're trying to avoid?

Reply:
k8s.io/utils is intentionally a leaf repo, so it should be OK IMO.

Reply:
yeah, that's fine

Reply:
Thanks to everyone for clarifying that! I'd wrongly assumed that was another repo sourced from k/k, but since I would actually need to add IPv4 functions to netutils to make it useful for these tests I've rolled back that change. I will try to find some time to file a PR on k/utils with those additions at some point, but it didn't seem reasonable to fit it in in time for this PR.